mlflow/models/cli.py

```diff
@@ -83,7 +83,9 @@ def predict(model_uri, input_path, output_path, content_type, json_format, no_co
 @click.option("--name", "-n", default="mlflow-pyfunc-servable",
               help="Name to use for built image")
 @cli_args.INSTALL_MLFLOW
-def build_docker(model_uri, name, install_mlflow):
+@click.option("--gunicorn_opts", "-g", default=None,
```

Comment:
Looking at this PR, we probably need to more broadly consider whether we allow passing arguments through to the container - for example, gunicorn opts are Python specific, and so to pass them you must know that your image contains a Python model, which breaks the idea that you can run the image created by `mlflow models build-docker` without worrying about the details of the embedded model. On the other hand, it does seem convenient to  to configure attributes of the server ([number of workers](http://docs.gunicorn.org/en/stable/settings.html#workers), [request size limits](http://docs.gunicorn.org/en/stable/settings.html#security), etc). cc @mateiz for an opinion here.

As @dbczumar pointed out to me, if we do allow passing arguments (like gunicorn opts) through to the container, they would probably be better passed as runtime arguments via `docker run`, rather than at image build time. To that end I'd suggest that we:
* Remove `gunicorn-opts` here
* Instead, update the [entry point helper method for the docker container](https://github.com/mlflow/mlflow/blob/a089618bfa509e71f8dfaffc0f42f13e71aa7942/mlflow/models/container/__init__.py#L39) to accept a command and optional positional args (e.g. so that we have `def _init(cmd, *args)`), and then pass the optional args through to `mlflow.models.container._serve()` to enable running with gunicorn opts.
* Add a file at e.g. mlflow/models/container/entry_point.py containing an `if __name__ == "__main__"` block that calls `mlflow.models.container._init(cmd, *sys.argv)`
* Update the [container entry point](https://github.com/mlflow/mlflow/blob/a089618bfa509e71f8dfaffc0f42f13e71aa7942/mlflow/pyfunc/backend.py#L115) to be e.g. `python -m mlflow.models.container.entry_point serve`.

Reply:
@lennon310 we discussed this today with the following conclusions:
* For gunicorn, it seems possible to specify arguments [via environment variables](http://docs.gunicorn.org/en/stable/settings.html)
* Thus, at image runtime you should already be able to configure your server via `docker run your-image -e GUNICORN_CMD_ARGS='your-desired-options'` (let me know if this isn't the case).
* At image build time, we can enhance `mlflow models build-docker` with e.g a repeatable `-e` option for specifying environment variables, and include them in the built image via the [ENV directive](https://docs.docker.com/engine/reference/builder/#env). These can then be overridden at runtime via the `docker run -e`, as described above.

Let me know what you think / if that makes sense!

Reply:
@smurching PR updated